File: segundos 4 lenguajes/c#/ProyectoCsharp/JugadorBase.cs
namespace ProyectoJugadores
{
    public class JugadorBase
    {
        public string Nombre { get; set; }
        public int Nivel { get; set; }

        public JugadorBase()
        {
            Nombre = "Juan";
            Nivel = 10;
        }

        public JugadorBase(string nombre, int nivel)
        {
            Nombre = nombre;
            Nivel = nivel;
        }

        public static JugadorBase Copy(JugadorBase j)
        {
            return new JugadorBase(j.Nombre, j.Nivel);
        }
    }
}

File: segundos 4 lenguajes/js/JugadorBase.js
export class JugadorBase {
    constructor(nombre = "Juan", nivel = 10, copia = null) {
        if (copia) {
            this.nombre = copia.nombre;
            this.nivel = copia.nivel;
        } else {
            this.nombre = nombre;
            this.nivel = nivel;
        }
    }

    atacar() {
        throw new Error("Método atacar() debe ser implementado.");
    }
}
